test_conformance/integer_ops/test_extended_bit_ops_extract.cpp

```diff
@@ -0,0 +1,288 @@
+{
+    typedef typename std::make_unsigned<T>::type unsigned_t;
+
+    assert(offset <= sizeof(T) * 8);
+    assert(count <= sizeof(T) * 8);
+    assert(offset + count <= sizeof(T) * 8);
+
+    unsigned_t base = static_cast<unsigned_t>(tbase);
+    unsigned_t result;
+
+    if (count == 0)
+    {
+        result = 0;
+    }
+    else
+    {
+        result = base << (sizeof(T) * 8 - offset - count);
+        result = result >> (sizeof(T) * 8 - count);
+    }
+
+    return result;
+}
+
+template <typename T, size_t N>
+static void
+calculate_reference(std::vector<typename std::make_unsigned<T>::type>& sref,
+                    std::vector<typename std::make_unsigned<T>::type>& uref,
+                    const std::vector<T>& base)
+{
+    sref.resize(base.size());
+    uref.resize(base.size());
+    for (size_t i = 0; i < base.size(); i++)
+    {
+        cl_uint offset = (i / N) / (sizeof(T) * 8 + 1);
+        cl_uint count = (i / N) % (sizeof(T) * 8 + 1);
+        if (offset + count > sizeof(T) * 8)
+        {
+            count = (sizeof(T) * 8) - offset;
+        }
+        sref[i] = cpu_bit_extract_signed(base[i], offset, count);
+        uref[i] = cpu_bit_extract_unsigned(base[i], offset, count);
+    }
+}
+
+static constexpr const char* kernel_source = R"CLC(
+__kernel void test_bitfield_extract(__global SIGNED_TYPE* sdst, __global UNSIGNED_TYPE* udst, __global TYPE* base)
+{
+    int index = get_global_id(0);
+    uint offset = index / (sizeof(BASETYPE) * 8 + 1);
+    uint count = index % (sizeof(BASETYPE) * 8 + 1);
+    if (offset + count > sizeof(BASETYPE) * 8) {
+        count = (sizeof(BASETYPE) * 8) - offset;
+    }
+    sdst[index] = bitfield_extract_signed(base[index], offset, count);
+    udst[index] = bitfield_extract_unsigned(base[index], offset, count);
+}
+)CLC";
+
+static constexpr const char* kernel_source_vec3 = R"CLC(
+__kernel void test_bitfield_extract(__global SIGNED_BASETYPE* sdst, __global UNSIGNED_BASETYPE* udst, __global BASETYPE* base)
+{
+    int index = get_global_id(0);
+    uint offset = index / (sizeof(BASETYPE) * 8 + 1);
+    uint count = index % (sizeof(BASETYPE) * 8 + 1);
+    if (offset + count > sizeof(BASETYPE) * 8) {
+        count = (sizeof(BASETYPE) * 8) - offset;
+    }
+    TYPE b = vload3(index, base);
+    SIGNED_TYPE s = bitfield_extract_signed(b, offset, count);
+    UNSIGNED_TYPE u = bitfield_extract_unsigned(b, offset, count);
+    vstore3(s, index, sdst);
+    vstore3(u, index, udst);
+}
+)CLC";
+
+template <typename T, size_t N>
+static int test_vectype(cl_device_id device, cl_context context,
+                        cl_command_queue queue)
+{
+    // Because converting from an unsigned type to a signed type is
+    // implementation-defined if the most significant bit is set until C++ 20,
+    // compute all reference results using unsigned types.
+    typedef typename std::make_unsigned<T>::type unsigned_t;
+
+    cl_int error = CL_SUCCESS;
+    int result = TEST_PASS;
```

Comment:
This is causing a build failure due to result being unused.

Reply:
Thanks, fixed!  I don't think this warning was enabled when I initially made these changes.